src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useState } from 'react';
+import { Loader } from '../components/Loader';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { PeopleTable } from '../components/PeopleTable/PeopleTable';
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const hasError = !isLoading && error;
+  const noPeople = !isLoading && people.length === 0;
+  const showTable = !isLoading && people.length > 0;
+
+  const peopleWithParents = people.reduce((newPeople: Person[], person) => {
+    const mother = people.find(mama => mama.name === person.motherName);
+    const father = people.find(papa => papa.name === person.fatherName);
+
+    const updatedPerson = {
+      ...person,
+      mother,
+      father,
+    };
+
+    newPeople.push(updatedPerson);
+
+    return newPeople;
+  }, []);
```

Comment:
The `peopleWithParents` computation involves a nested loop where for each person, it searches the entire list to find their parents. This can be inefficient for large datasets. Consider using a map or object to store people by name for faster lookup.